indra/llcharacter/lljoint.cpp

```diff
@@ -1050,5 +1037,24 @@ void LLJoint::setSkinOffset( const LLVector3& offset )
 	mSkinOffset = offset;
 }
 
+//-----------------------------------------------------------------------------
+// clampRotation()
+//-----------------------------------------------------------------------------
+void LLJoint::clampRotation(LLQuaternion old_rot, LLQuaternion new_rot)
+{
+	LLVector3 main_axis(1.f, 0.f, 0.f);
+
+	for (LLJoint* joint : mChildren)
+	{
+		if (joint->isAnimatable())
+		{
+			main_axis = joint->getPosition();
+			main_axis.normVec();
+			// only care about first animatable child
+			break;
+		}
+	}
+}
+
```

Comment:
`clampRotation()` is bad: it doesn't do what it claims, and it is also not used anywhere.  This is probably cruft I had removed when looking over **lljoint.cpp**.  I don't remember doing it, but it is what I would do now.

Reply:
Re-removed